### ISSUE TITLE
Use xraydb instead of xraylib

### ISSUE THREAD

**Issue by joefowler** [usnistgov/mass#209]
**[Original report](https://bitbucket.org/joe_fowler/mass/issues/209) by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

The `mass.materials` package uses `xraylib` for its x-ray data. I think that we’d be happier with [XrayDB](https://xraypy.github.io/XrayDB/index.html). Its data is not different except in certain small details. \(“small”??\) In favor of xraydb:

* It’s a much simpler piece of code: 2800 lines of pure Python plus a 10 MB SQLite database. It will be easier to install. \(no compiling! no challenging installs on the CI systems!\)
* It’s actively maintained \(27 commits in the last 6 months\) and by Matt Newville of U Chicago, whom we trust. \(To be clear, [xraylib](https://github.com/tschoonj/xraylib) is [actively maintained](https://github.com/tschoonj/xraylib/commits/master) and trusted, too\).
* Its interface is simpler and considerably more pythonic. \(See example steps 124-126, below.\)
* Interface is a bit more elegant in handling compounds. \(See example steps 127-128.\)
* Interface accepts and returns vector arguments where xraylib doesn’t. \(See example steps 129-131.\)

‌

```text
In [124]: xraylib.ElementDensity(xraylib.SymbolToAtomicNumber('Al'))
Out[124]: 2.6989

In [125]: xraydb.atomic_density("Al")
Out[125]: 2.7

In [126]: xraydb.atomic_density(13)
Out[126]: 2.7

In [127]: xraydb.material_mu("kapton", [3000, 4000, 5000])
Out[127]: array([168.31261592,  71.1876784 ,  36.28036915])

In [128]: xraylib.GetCompoundDataNISTByName("Kapton Polyimide Film")
Out[128]: 
{'name': 'Kapton Polyimide Film',
 'nElements': 4,
 'density': 1.42,
 'Elements': (1, 6, 7, 8),
 'massFractions': (0.026362, 0.691133, 0.07327, 0.209235)}

In [129]: [xraylib.CS_Total(13, x) for x in [3, 4, 5]]
Out[129]: [788.1123227920823, 360.47510057473966, 193.3977103779502]

In [130]: np.array([xraylib.CS_Total(13, x) for x in [3, 4, 5]])*xraylib.ElementDensity(13)
Out[130]: array([2127.03634798,  972.88624894,  521.96108054])

In [131]: xraydb.material_mu("Al", [3000, 4000, 5000])
Out[131]: array([2127.90327154,  973.28277155,  522.17381802])
```

I like how you can pass in atomic number or element name and get the answer without fussing over conversion to the “right” form of input. Also, it’s cool that xraylib computes the mass fractions, but xraydb does even better: it folds the mass fractions, the density, and the x-ray cross sections of the elements all together to give you what you really want.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Fixed in 04627d7 and v0.7.8, back in November 2021. I'm not sure how we failed to close this issue at the time, but it's done now.

**Comment by joefowler** [comment]
**Original comment by Joseph Fowler (Bitbucket: [joe_fowler](https://bitbucket.org/joe_fowler), ).**

----------------------------------------

Removing milestone: v0.7 (automated comment)
